bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -115,6 +115,20 @@ case class Proposal(meta:Meta,
 
   def resetObservationMeta:Proposal = copy(observations = observations.map(_.copy(meta = None)))
 
+  // Calculate the program time as a ratio of total time.
+  private def timeSum(extract: Observation => Option[TimeAmount]): TimeAmount =
+    TimeAmount.sum(observations.map(extract).flatten)
+
+  def programTimeHours: TimeAmount = timeSum(_.progTime)
+  def partnerTimeHours: TimeAmount = timeSum(_.partTime)
```

Comment:
I was actually just thinking about that myself. This class should be fully immutable, so there's no reason why we can't make them all `lazy val`s.